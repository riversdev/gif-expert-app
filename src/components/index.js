export * from './AddCategorie'
export * from './GifGrid'
export * from './GifItem'