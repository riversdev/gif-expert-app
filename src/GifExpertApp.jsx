import { useState } from 'react'
import { AddCategory, GifGrid } from './components'

export const GifExpertApp = () => {
    const [categories, setCategories] = useState(['Naruto'])
    const onAddCategory = category => {
        // if (categories.includes(categorie)) return
        if (categories.find(x => x.toLowerCase() === category.toLowerCase())) return

        setCategories([
            category,
            ...categories,
        ])
    }

    return (
        <>
            <h1>Gif expert app by Rivers</h1>
            <AddCategory onAddCategory={onAddCategory} />
            {categories.map(category =>
                <GifGrid key={category} category={category} />
            )}
        </>
    )
}