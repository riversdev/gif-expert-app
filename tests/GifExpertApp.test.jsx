import { render, screen } from '@testing-library/react'
import { GifExpertApp } from '../src/GifExpertApp'

describe('GifExpertApp tests', () => {
    test('should first', () => {
        const { container } = render(<GifExpertApp />)

        expect(screen.getByRole('heading', { level: 1 })).toBeTruthy()
        expect(screen.getAllByRole('heading', { level: 1 }).length).toBe(1)
        expect(container).toMatchSnapshot()

        // screen.debug()
    })
})