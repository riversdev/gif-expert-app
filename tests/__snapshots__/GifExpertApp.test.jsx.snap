// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifExpertApp tests should first 1`] = `
<div>
  <h1>
    Gif expert app by Rivers
  </h1>
  <form
    aria-label="gifsForm"
  >
    <input
      placeholder="Buscar gifs"
      type="text"
      value=""
    />
  </form>
  <h3>
    Naruto
  </h3>
  <h2>
    Cargando...
  </h2>
  <div
    class="card-grid"
  />
</div>
`;
